### DISCUSSION TITLE
SQL Joins of multiple documents?

### DISCUSSION THREAD

**Discussion by genachka** [ravendb/ravendb#14762] [Ideas]
Coming from using Couchbase since about version 2 I'm used to working with Map Reduce but must admit when they added their N1QL language to do SQL constructs and when they added Joins it got a lot easier to maintain the database and the Java client code we use. RQL in many respects feels like SQL. It would be amazing if doing multiple document joins could simply be done within RQL without having to do the extra leg work. Sure I understand that I'm allowing the database engine figure out the best paths for me, but in many use cases it's perfect and in most cases performance doesn't suffer in my experience. Plus the learning curve of adding other folks is reduced significantly. Not saying Map Reduce needs to go away, but an option of not having to solely use it that was like a SQL Join would be ideal.

**Comment by ayende** [comment]
Have you see the ability to call `load` and `include` during the query?

https://ravendb.net/docs/article-page/5.4/csharp/indexes/indexing-related-documents
https://ravendb.net/docs/article-page/5.4/csharp/indexes/querying/basics#query-flow
https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/loading-entities#load-with-includes

For example:

```
from Orders as o
load o.Company as c
select c.Name as CompanyName, o.OrderedAt

```
